File: client/src/app/modules/miscellaneous/services/gift.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { Gift } from '../../miscellaneous/model/Gift';

@Injectable({
  providedIn: 'root'
})
export class GiftService extends AppRestService<Gift>{

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getGifts() {
    return this.getListCached(new HttpParams(), "gifts");
  }

  addOrUpdateGift(gift: Gift) {
    this.clearListCache(new HttpParams(), "gifts");
    return this.addOrUpdate(new HttpParams(), "gift", gift, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/miscellaneous/service/GiftService.java
package com.jss.osiris.modules.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.miscellaneous.model.Gift;

public interface GiftService {
    public List<Gift> getGifts();

    public Gift getGift(Integer id);
	
	 public Gift addOrUpdateGift(Gift gift);
}
